pkg/cmd/target/drop/command_test.go

```diff
@@ -0,0 +1,213 @@
+			Gardens: []config.Garden{{
+				Name:       gardenName,
+				Kubeconfig: gardenKubeconfig,
+			}},
+		}
+
+		// user has already targeted a garden and project
+		currentTarget := target.NewTarget(gardenName, projectName, "", "")
+
+		// garden cluster contains the targeted project
+		project := &gardencorev1beta1.Project{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: projectName,
+			},
+			Spec: gardencorev1beta1.ProjectSpec{
+				Namespace: pointer.String("garden-prod1"),
+			},
+		}
+
+		fakeGardenClient := fake.NewClientBuilder().WithObjects(project).Build()
+
+		// setup command
+		targetProvider := internalfake.NewFakeTargetProvider(currentTarget)
+		clientProvider := internalfake.NewFakeClientProvider()
+		clientProvider.WithClient(gardenKubeconfig, fakeGardenClient)
+
+		factory := internalfake.NewFakeFactory(cfg, nil, clientProvider, nil, targetProvider)
+		cmd := NewCommand(factory, NewOptions(streams), &target.DynamicTargetProvider{})
+
+		// run command
+		Expect(cmd.RunE(cmd, []string{"project"})).To(Succeed())
+		Expect(out.String()).To(ContainSubstring("Successfully dropped targeted project %q\n", projectName))
+
+		currentTarget, err := targetProvider.Read()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(currentTarget.GardenName()).To(Equal(gardenName))
+		Expect(currentTarget.ProjectName()).To(BeEmpty())
+	})
+
+	It("should be able to drop targeted seed", func() {
+		streams, _, out, _ := util.NewTestIOStreams()
+
+		gardenName := "mygarden"
+		seedName := "myseed"
+		gardenKubeconfig := ""
+		cfg := &config.Config{
+			Gardens: []config.Garden{{
+				Name:       gardenName,
+				Kubeconfig: gardenKubeconfig,
+			}},
+		}
+
+		// user has already targeted a garden and seed
+		currentTarget := target.NewTarget(gardenName, "", seedName, "")
+
+		// garden cluster contains the targeted seed
+		seed := &gardencorev1beta1.Seed{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: seedName,
+			},
+			Spec: gardencorev1beta1.SeedSpec{
+				SecretRef: &corev1.SecretReference{
+					Namespace: "garden",
+					Name:      seedName,
+				},
+			},
+		}
+
+		fakeGardenClient := fake.NewClientBuilder().WithObjects(seed).Build()
+
+		// setup command
+		targetProvider := internalfake.NewFakeTargetProvider(currentTarget)
+		clientProvider := internalfake.NewFakeClientProvider()
+		clientProvider.WithClient(gardenKubeconfig, fakeGardenClient)
+
+		factory := internalfake.NewFakeFactory(cfg, nil, clientProvider, nil, targetProvider)
+		cmd := NewCommand(factory, NewOptions(streams), &target.DynamicTargetProvider{})
+
+		// run command
+		Expect(cmd.RunE(cmd, []string{"seed"})).To(Succeed())
+		Expect(out.String()).To(ContainSubstring("Successfully dropped targeted seed %q\n", seedName))
+
+		currentTarget, err := targetProvider.Read()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(currentTarget.GardenName()).To(Equal(gardenName))
+		Expect(currentTarget.SeedName()).To(BeEmpty())
+	})
+
+	It("should be able to drop targeted shoot", func() {
```

Comment:
add test when seed is targeted and project is dropped

Reply:
There is already a test for this: `should unset deeper target levels when dropping project` this test checks, that both seed and shoot get resetted when dropping the project.

Reply:
ok 👍